api/v1/mba.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 QCT (Quanta Cloud Technology). All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package v1
+
+import (
+	"github.com/emicklei/go-restful"
+	"github.com/intel/rmd/lib/mba"
+	"github.com/intel/rmd/lib/proc"
+)
+
+// MbaInfo represents Mba API
+type MbaInfo struct {
+	Mba     bool `json:"mba"`
+	MbaOn   bool `json:"mba_enable,omitempty"`
+	MbaStep int  `json:"mba_step,omitempty"`
+	MbaMin  int  `json:"mba_min,omitempty"`
+}
+
+// Register handlers
+func (c *MbaInfo) Register(container *restful.Container) {
+	flag, err := proc.IsMbaAvailiable()
```

Comment:
Why RMD has to check resctrl info every REST call?
Resctrl info will not change until we change CPU or reboot with another kernel parameters.

Ideally the resctrl info should collected during the initialization and cached by RMD itself.
Then, RMD could respond user queries without additional system call.

Because there is no additional system call or other operation during REST call,
there is no error handling could be implemented.

Unit tests should be added.
I'll add them later.

Reply:
It is cleaner to have it within every REST call as RMD does not cache any information, except policies in its DB. We can expect remounting of resctrl, though not advised while RMD is still running. 
Secondly we do not expect the application to crash for one GET operation failure that has been done in Register.

IsMbaAvailable and GetMbaInfo both return errors. Please propagate these to GET API to report an Internal Server Error.

Please submit code changes with unit tests. It makes the review process easier.

Reply:
I'm not sure that if there is any difference between our systems.
RMD seems like keeping a lot of information in my survey.

Here are 2 examples:

A. RMD store(cache) the workload information, the information between RMD and resctrl are different
   (Are them the policies you mention?)
1. Run RMD in a fresh environment, there is no workload/COS in RMD/resctrl
2. Create a workload via POST /workloads
3. umount resctrl
4. Query workloads via GET /workloads,
   and then you'll see that the response includes the workload you just created

B. RMD store(cache) a lot of cache information:
1. Run RMD in a fresh environment, there is no workload/COS in RMD/resctrl
2. Query cache information via GET /cache/l3
3. umount resctrl
4. Query cache information via GET /cache/l3,
   and then you'll get a response includes cache information without any Internal Server Error.

C. The data in DB are useless if we talk about the "remounting"
1. Run RMD in a fresh environment, there is no workload/COS in RMD/resctrl
2. Create a workload via POST /workloads
3. Kill RMD process, umount resctrl
4. mount resctrl, Run RMD again
4. Query workloads via GET /workloads,
   and then you'll see there is no any workload in response.

I'm disoriented.
I don't know what's about "RMD does not cache any information" you mention.
Some data is stored in DB, but it seems like useless if we reboot the machine.

How about let me make a "cache version" of /mba, and then you imporve it to a "non-cache version"?